spec/models/work_spec.rb

```diff
@@ -463,38 +467,99 @@
         "http://efiction-site.com/viewstory.php?sid=123" => "http://efiction-site.com/viewstory.php?sid=456",
         "http://www.foo.com/i-am-something" => "http://foo.com/i-am-something/else"
       }.each do |import_url, find_url|
-        work = create(:work, imported_from_url: import_url)
+        work = create(:work)
+        work.import_url!(import_url)
         expect(Work.find_by_url(find_url)).to_not eq(work)
         work.destroy
       end
     end
 
     it "should find works imported with irrelevant query parameters" do
-      work = create(:work, imported_from_url: "http://lj-site.com/thing1?style=mine")
+      work = create(:work)
+      work.import_url!("http://lj-site.com/thing1?style=mine")
       expect(Work.find_by_url("http://lj-site.com/thing1?style=other")).to eq(work)
       work.destroy
     end
 
     it "finds works imported with HTTP protocol and irrelevant query parameters" do
-      work = create(:work, imported_from_url: "http://lj-site.com/thing1?style=mine")
+      work = create(:work)
+      work.import_url!("http://lj-site.com/thing1?style=mine")
       expect(Work.find_by_url("https://lj-site.com/thing1?style=other")).to eq(work)
       work.destroy
     end
 
     it "finds works imported with HTTPS protocol and irrelevant query parameters" do
-      work = create(:work, imported_from_url: "https://lj-site.com/thing1?style=mine")
+      work = create(:work)
+      work.import_url!("https://lj-site.com/thing1?style=mine")
       expect(Work.find_by_url("http://lj-site.com/thing1?style=other")).to eq(work)
       work.destroy
     end
 
     it "gets the work from cache when searching for an imported work by URL" do
       url = "http://lj-site.com/thing2"
-      work = create(:work, imported_from_url: url)
-      expect(Rails.cache.read(Work.find_by_url_cache_key(url))).to be_nil
+      work = create(:work)
+      work.import_url!(url)
+      expect(Rails.cache.read(WorkUrl.find_by_url_cache_key(url))).to be_nil
       expect(Work.find_by_url(url)).to eq(work)
-      expect(Rails.cache.read(Work.find_by_url_cache_key(url))).to eq(work)
+      expect(Rails.cache.read(WorkUrl.find_by_url_cache_key(url))).to eq(work)
       work.destroy
     end
+
+    describe "legacy – to be removed eventually – cf AO3-6979" do
+      it "should find imported works with various URL formats" do
```

Comment:
For this (and the other modified cases), can you reword them to avoid should? https://www.betterspecs.org/#should